src/em/prov/easyconnect/ec_ctrl_configurator.cpp

```diff
@@ -1116,7 +937,187 @@ std::pair<uint8_t *, size_t> ec_ctrl_configurator_t::create_recfg_auth_confirm(s
     return std::make_pair(reinterpret_cast<uint8_t*>(frame), EC_FRAME_BASE_SIZE + attribs_len);
 }
 
-std::pair<uint8_t *, size_t> ec_ctrl_configurator_t::create_config_response()
+cJSON *ec_ctrl_configurator_t::complete_config_obj(cJSON *base, ec_connection_context_t& conn_ctx, dpp_config_obj_type_e config_obj_type)
 {
-    return {};
+    if (base == nullptr) {
+        printf("%s:%d: Based a nullptr base Configuration object\n", __func__, __LINE__);
+        return nullptr;
+    }
+    switch(config_obj_type) {
```

Comment:
Since all of the fields are the same except for `groups`, these field creations could be moved outside of the switch statement.

It would change things substantially (due to a need for a check on if the type is valid before hand) but it could avoid people reading in the future trying to play a game of spot the difference. 

You could put a switch at the top to the filtering if chosen:

```suggestion
    std::vector<std::unordered_map<std::string, std::string>> groups;
    switch(config_obj_type) {
        case dpp_config_obj_type_e::dpp_config_obj_bsta: {
           groups = {
                {{"groupID", "mapNW"}, {"netRole", "mapBackhaulSta"}}
            };
            break;
        }
        case dpp_config_obj_type_e::dpp_config_obj_ieee1905: {
           groups = {
                {{"groupID", "mapNW"}, {"netRole", "mapAgent"}}
            };
            break;
        }
        default:
            // Error
            return NULL;
    }

```